helm-chart/binderhub/values.yaml

```diff
@@ -39,48 +40,63 @@ config:
 
 extraConfig: {}
 
-# have to set cors.allowOrigin twice:
-# once in top-level cors.allowOrigin,
-# and again in jupyterhub.hub.extraConfigMap.cors.allowOrigin
-
-# Using YAML anchors, `&cors` for the first appearance and `*cors` for subsequent
-# appearances, allows us to remove redundancy in this (BinderHub) `values.yaml`.
-# The anchors do not extend beyond this file.
-# As such, users must set `cors` separately for their own user `values.yaml` for notebooks.
-# `cors` will be set separately in the user `values.yaml` and binderhub`values.yaml`.
-# The same anchor pattern (`&cors`, `*cors`) can be used in the user `values.yaml`.
-
-cors: &cors
-  allowOrigin:
+# Two bits of config need to be set to fully enable cors.
+# config.BinderHub.cors_allow_origin controls the allowed origins for the
+# binderhub api, and jupyterhub.hub.config.BinderSpawner.cors_allow_origin
+# controls the allowed origins for the spawned user notebooks. You most
+# likely want to set both of those to the same value.
 
 jupyterhub:
+  # Deprecated values, kept here so we can provide useful error messages
+  custom:
+    cors: {}
   cull:
     enabled: true
     users: true
-  custom:
-    cors: *cors
-    binderauth_enabled: false
   rbac:
     enabled: true
   hub:
     config:
       JupyterHub:
         authenticator_class: nullauthenticator.NullAuthenticator
+      BinderSpawner:
+        auth_enabled: false
```

Comment:
A quick clarification regarding this: Will this break downstream packages which define custom Spawners and not use BinderSpawner?

Reply:
Depends, what do they inherit from? If they inherit from BinderSpawner, everything should be fine I think.

Reply:
This config is like writing `c.BinderSpawner.auth_enabled = False` and will influence a MyCustomBinderSpawner class as well as long as it inherits from BinderSpawner

Reply:
Thanks! We [inherit from `KubeSpawner`](https://github.com/gesiscss/persistent_binderhub/blob/2c2f87ac567d2c46b8e8848d76700e3c1da51a63/persistent_binderhub/files/jupyterhub/persistent_bhub_config.py#L39) so need to fix that up for Persistent BinderHub :)